lib/modules/platform/codecommit/index.ts

```diff
@@ -0,0 +1,716 @@
+      'you tick the rebase/retry checkbox',
+      'rename PR to start with "rebase!"'
+    )
+    .replace(regEx(/<\/?summary>/g), '**')
+    .replace(regEx(/<\/?details>/g), '')
+    .replace(regEx(`\n---\n\n.*?<!-- rebase-check -->.*?\n`), '')
+    .replace(regEx(/\]\(\.\.\/pull\//g), '](../../pull-requests/')
+    .replace(
+      regEx(/(?<hiddenComment><!--renovate-(?:debug|config-hash):.*?-->)/g),
+      '[//]: # ($<hiddenComment>)'
+    );
+}
+
+export async function getJsonFile(
+  fileName: string,
+  repoName?: string,
+  branchOrTag?: string
+): Promise<any | null> {
+  const raw = await getRawFile(fileName, repoName, branchOrTag);
+  return raw ? JSON5.parse(raw) : null;
+}
+
+export async function getRawFile(
+  fileName: string,
+  repoName?: string,
+  branchOrTag?: string
+): Promise<string | null> {
+  const fileRes = await client.getFile(
+    repoName ?? config.repository,
+    fileName,
+    branchOrTag
+  );
+  if (!fileRes?.fileContent) {
+    return null;
+  }
+  const buf = Buffer.from(fileRes.fileContent);
+  return buf.toString('utf-8');
+}
+
+/* istanbul ignore next */
+export function getRepoForceRebase(): Promise<boolean> {
+  return Promise.resolve(false);
+}
+
+export async function createPr({
+  sourceBranch,
+  targetBranch,
+  prTitle: title,
+  prBody: body,
+}: CreatePRConfig): Promise<Pr> {
+  const description = smartTruncate(sanitize(body), 10239);
+
+  const prCreateRes = await client.createPr(
+    title,
+    sanitize(description),
+    sourceBranch,
+    targetBranch,
+    config.repository
+  );
+
+  if (
+    !prCreateRes.pullRequest?.title ||
+    !prCreateRes.pullRequest?.pullRequestId
+  ) {
+    throw new Error('Could not create pr, missing PR info');
+  }
+
+  return {
+    number: Number.parseInt(prCreateRes.pullRequest.pullRequestId),
+    state: PrState.Open,
+    title: prCreateRes.pullRequest.title,
+    sourceBranch,
+    targetBranch,
+    sourceRepo: config.repository,
+  };
+}
+
+export async function updatePr({
+  number: prNo,
+  prTitle: title,
+  prBody: body,
+  state,
+}: UpdatePrConfig): Promise<void> {
+  logger.debug(`updatePr(${prNo}, ${title}, body)`);
+
+  if (body) {
+    await client.updatePrDescription(
+      `${prNo}`,
+      smartTruncate(sanitize(body), 10239)
+    );
+  }
+
+  if (title) {
+    await client.updatePrTitle(`${prNo}`, title);
+  }
+
+  const prStatusInput =
+    state === PrState.Closed
+      ? PullRequestStatusEnum.CLOSED
+      : PullRequestStatusEnum.OPEN;
+  try {
+    await client.updatePrStatus(`${prNo}`, prStatusInput);
+  } catch (err) {
+    // do nothing, it's ok to fail sometimes when trying to update from open to open or from closed to closed.
+  }
+}
+
+// Auto-Merge not supported currently.
+/* istanbul ignore next */
+export async function mergePr({
+  branchName,
+  id: prNo,
+  strategy,
+}: MergePRConfig): Promise<boolean> {
+  logger.debug(`mergePr(${prNo}, ${branchName!})`);
+
+  const prOut = await client.getPr(`${prNo}`);
```

Comment:
can this throw? we should avoid throwing from this function

Reply:
auto merge is not supported, wasn't tested e2e , irrelevant